.k8s/K8s-Deployment.md

```diff
@@ -0,0 +1,62 @@
+# Kubernetes Deployment
+## Deployment
+### The Tool
+To work with a K8s cluster you need to install Kubectl, a tool for controlling your K8s deployment.
+[Kubectl installation guide](https://kubernetes.io/docs/tasks/tools/install-kubectl/)
+
+The `docker` command is the Docker Swarm equivalent to `kubectl`.
+
+By default the deploy-k8s.groovy deployment script will deploy to a local Minikube instance. To deploy to a remote
+Kubernetes cluster, you will need to add the the Docker registry environment variable `K8S_DOCKER_REGISTRY`. You
+will also need to change `kubectl`'s context to the remote one with `kubectl config use-context <remote-context>`
+
+#### Optional:
+Install MiniKube, a single node Kubernetes cluster that runs locally in a VM or the host machine. Deploys a lightweight single-node k8s cluster.
+Note: Run `eval $(minikube docker-env)` to point to Minikube's docker daemon. You need to do this when you build images
+for local deployment.
+
+When starting up [Minikube](https://kubernetes.io/docs/tasks/tools/install-minikube/), make sure your vm-driver is the one bundled with
+VirtualBox. Dnsmasq doesn't play nice with the standalone hyperkit.
+
+### Before Deploying
+
+The following is for informational pruposes only. The `./gradlew deployK8s` gradle task or the `groovy deploy-k8s`
+script will do the following steps automatically.
+
+Kubernetes manifests do not support adding config files directly in the manifest via a file path. Instead we need to use
+a resource called ConfigMaps.
+
+ConfigMaps are how we pass configurations to the containers. These can be created through literals or files. The only
+issue is that these can't be created in a manifest based off a file like we used to do with Docker Compose. Now we have
+to created them manually using the following commands:
```

Comment:
Having looked at `store_config.yml`, and I imagine `transform_config.yml` looks much the same. You could make a config map as part of the manifest and putting the contents inline. I don't believe there's nothing sensitive in those two to worrying about committing to VC.

Reply:
I think I'll do that once more services switch to K8s.